### PR TITLE
Add tests for internal/pkg/instance

### PR DESCRIPTION
**Pull request by cclerget**
**Description of the Pull Request (PR):**

- #2559 

**Before submitting a PR, make sure you have done the following:**

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/master/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

File: internal/pkg/instance/instance_linux_test.go [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+	"github.com/sylabs/singularity/internal/pkg/test"
+)
+
+const testSubDir = "testing"
+
+func TestProcName(t *testing.T) {
+	test.DropPrivilege(t)
+	defer test.ResetPrivilege(t)
+
+	test := ProcName("test", "test")
```
Comment (mem) [inline internal/pkg/instance/instance_linux_test.go]: here `test` is a new variable; on line 20 `test` is a package selector. To avoid the ambiguity it's probably better to name the variable something else, like `actual` to reflect that this is the actual value obtained from calling the function under test.

Also, without looking at the internals of ProcName, I would expect that there are at least four interesting cases to test:

```
    ProcName("str1", "str2")
    ProcName("str1", "")
    ProcName("", "str1")
    ProcName("", "")
```

- two different strings (to test that the correct one goes in the correct place)
- second string empty
- first string empty
- both strings empty

File: internal/pkg/instance/instance_linux_test.go [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+		},
+		{
+			input:  "instance:///test",
+			output: "/test",
+		},
+	}
+	for _, e := range tests {
+		o := ExtractName(e.input)
+		if o != e.output {
+			t.Errorf("unexpected match, got %s instead of %s", o, e.output)
```
Comment (mem) [inline internal/pkg/instance/instance_linux_test.go]: "unexpect result" maybe?

File: internal/pkg/instance/instance_linux_test.go [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+			expectFailure: false,
+		},
+		{
+			name:          "test/123",
+			expectFailure: true,
+		},
+	}
+	for _, e := range tests {
+		path, err := GetDirUnprivileged(e.name, testSubDir)
+		if err != nil && !e.expectFailure {
```
Comment (mem) [inline internal/pkg/instance/instance_linux_test.go:189]: there's also the case of:

    if err == nil && e.expectFailure {
        t.Errorf("...")
    }

the complete set could be expressed as:

    switch {
    case err == nil && e.expectFailure:
        t.Errorf(...)
    case err != nil && !e.expectFailure:
        t.Errorf(...)
    case err != nil && e.expectFailure:
        // everything OK
    case path != e.path:
        t.Errorf(...)
     }
```

File: internal/pkg/instance/instance_linux_test.go [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+			path:          filepath.Join(instancePath, "test"),
+			expectFailure: false,
+		},
+		{
+			name:          "test/123",
+			expectFailure: true,
+		},
+	}
+	for _, e := range tests {
+		path, err := GetDirPrivileged(e.name, testSubDir)
```
Comment (mem) [inline internal/pkg/instance/instance_linux_test.go:220]: same here

File: internal/pkg/instance/instance_linux_test.go [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+
+func TestAdd(t *testing.T) {
+	test.EnsurePrivilege(t)
+
+	for _, e := range instanceTests {
+		var err error
+		var file *File
+
+		file, err = Add(e.name, e.privileged, testSubDir)
+		if err != nil && !e.expectFailure {
```
Comment (mem) [inline internal/pkg/instance/instance_linux_test.go:271]: something similar here. You can also test for an unexpect file == nil, I guess.

File: internal/pkg/instance/instance_linux_test.go [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+
+func TestGet(t *testing.T) {
+	test.EnsurePrivilege(t)
+
+	for _, e := range instanceTests {
+		var err error
+		var file *File
+
+		file, err = Get(e.name, testSubDir)
+		if err != nil && !e.expectFailure {
```
Comment (mem) [inline internal/pkg/instance/instance_linux_test.go:317]: and here, too.

**Review by mem — CHANGES_REQUESTED** [review]

**Review by mem — APPROVED** [review]
thanks!
